lib/supabase.ts

```
@@ -55,18 +54,21 @@ if (!supabaseUrl || !supabaseAnonKey) {
 
-export const supabase = createClient(
-  supabaseUrl || 'https://placeholder.supabase.co',
-  supabaseAnonKey || 'placeholder-key',
-  {
-    auth: {
-      persistSession: true,
-      autoRefreshToken: true,
+export function Supabase() {
+  const { session } = useSession()
+
+  return createClient(supabaseUrl!, supabaseAnonKey!, {
```

Biome (2.1.2)

[error] This hook is being called indirectly and conditionally, but all hooks must be called in the exact same order in every component render.

This is the call path until the hook.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)